src/create-release.js

```diff
@@ -161,6 +161,8 @@ async function run() {
     const body = core.getInput('body', { required: false });
     const draft = core.getInput('draft', { required: false }) === 'true';
 
+    const commitish = core.getInput('commitish', { required: false }) || context.sha;
```

Comment:
it is debatable whether or not we want to default to `context.sha` or just leave it `undefined` if not provided by the user.

When we leave it `undefined` we don't impact current usages of this action

Reply:
Most definitely should default to `context.sha` as that is the only way that makes sense. The current implementation (without this PR) where the created release doesn't match the action it's ran from is absolutely unintuitive. 

So for example, if you run a build action to compile binaries and release them from other branch, or from an earlier commit, the release itself would be tagged to the master branch and latest commit while the binaries would be from a different one and thus the release information and the source code wouldn't match the compiled binaries. I highly doubt that there is any user out there who *intends* to do this.

For the less popular use cases where the action is not ran against a commit, like from cron jobs, `commit.sha == undefined` anyway so that wouldn't change.

I think it's quite obviously intuitive that when you run an action from whatever branch or commit, that the release should match that commit unless specified otherwise. That's exactly how you did it, and that's exactly how it should work. I don't think it's even debatable, as I cannot imagine a counter-argument against the logic how you currently have it.

I'd say the correct way of making sure even the most weirdest of configurations wouldn't get affected whatsoever, is not to use the old flawed logic of `undefined` but rather bump the version tag from v2 to v3.

Reply:
So a TL;DR; version for the maintainer:
Change nothing, keep this PR exactly as it is now, and when you merge this just bump the version tag to v3 afterwards, that's it.